packages/salesforcedx-vscode-core/src/commands/forceSourceRetrieveCmp.ts

```diff
@@ -0,0 +1,177 @@
+    case 'LightningComponentBundle':
+      suffixes = LWC_DEFINITION_FILE_EXTS;
+      break;
+    case 'AuraDefinitionBundle':
+      suffixes = AURA_DEFINITION_FILE_EXTS;
+      break;
+    default:
+      suffixes = [`.${typeNode.suffix!}`];
+  }
+  return suffixes.map(suffix => `${suffix!}-meta.xml`);
+}
+
+export async function forceSourceRetrieveCmp(componentNode: BrowserNode) {
+  const typeNode =
+    componentNode.parent!.type === NodeType.Folder
+      ? componentNode.parent!.parent!
+      : componentNode.parent!;
+  const typeName = typeNode.fullName;
+  const dirName = typeNode.directoryName!;
+  const componentName = componentNode.fullName;
+  const label =
+    componentNode.parent!.type === NodeType.Folder
+      ? componentName.substr(componentName.indexOf('/') + 1)
+      : componentName;
+  const fileExts = generateSuffix(typeNode, typeName);
+
+  const sourcePathStrategy = BUNDLE_TYPES.has(typeName)
+    ? new BundlePathStrategy()
+    : new DefaultPathStrategy();
+
+  const metadataArg = `${typeName}:${componentName}`;
+  const executor = new ForceSourceRetrieveExecutor(metadataArg);
+
+  const commandlet = new SfdxCommandlet(
+    workspaceChecker,
+    parameterGatherer,
+    executor,
+    new FilePathExistsChecker(
+      fileExts,
+      sourcePathStrategy,
+      componentName,
+      dirName,
+      label
+    )
+  );
+  await commandlet.run();
+}
+
+export class FilePathExistsChecker implements PostconditionChecker<{}> {
+  private fileExts: string[];
+  private sourcePathStrategy: SourcePathStrategy;
+  private componentName: string;
+  private dirName: string;
+  private label: string;
+
+  public constructor(
+    fileExts: string[],
+    sourcePathStrategy: SourcePathStrategy,
+    componentName: string,
+    dirName: string,
+    label: string
+  ) {
+    this.fileExts = fileExts;
+    this.sourcePathStrategy = sourcePathStrategy;
+    this.componentName = componentName;
+    this.dirName = dirName;
+    this.label = label;
+  }
+
+  public async check(): Promise<ContinueResponse<{}> | CancelResponse> {
+    const files = await vscode.workspace.findFiles(
+      await this.createFilesGlob(this.dirName, this.componentName)
+    );
+    if (files.length === 0) {
+      return { type: 'CONTINUE', data: {} };
+    } else {
+      const overwrite = await notificationService.showWarningMessage(
+        nls.localize('warning_prompt_cmp_file_overwrite', this.label),
+        nls.localize('warning_prompt_overwrite_confirm'),
+        nls.localize('warning_prompt_overwrite_cancel')
+      );
+      if (overwrite === nls.localize('warning_prompt_overwrite_confirm')) {
+        return { type: 'CONTINUE', data: {} };
+      }
+    }
+    return { type: 'CANCEL' };
+  }
+
+  private async createFilesGlob(
+    dirName: string,
+    fileName: string
+  ): Promise<string> {
+    const packageDirectories = await SfdxPackageDirectories.getPackageDirectoryPaths();
+    const basePath = path.join(
+      packageDirectories[0],
```

Comment:
`packageDirectories[0]` is always the path set as default in `sfdx-project.json`'s `packageDirectories` array . There is a problem in this method because it looks like the cli retrieve command is not respecting setting `default: true` to a path different than `force-app`.

![Screen Shot 2019-07-22 at 7 33 54 PM](https://user-images.githubusercontent.com/1041842/61678624-3edad700-acb8-11e9-8d03-d2cf961efdd0.png)

Reply:
Do you have anything else in your `sfdx-project.json`? I'm trying to reproduce what you're seeing, but I haven't been able to yet.
<img width="818" alt="Screen Shot 2019-07-22 at 8 08 57 PM" src="https://user-images.githubusercontent.com/21326913/61679931-b27ee300-acbc-11e9-8625-d320ea2cae5c.png">

Reply:
I was testing why a brand new project I created using `SFDX: Create Project with Manifest`. I only added new `path` elements under `packageDirectories`. I basically edited the `sfdx-project.json` folder to add new paths and change which one is the default. After saving the change I went to the org browser and did a retrieve. It looks like the cli takes some time to catch that there was a change on which package directory is now set to default. Might be a caching issue at the cli level.

Reply:
Ah okay, so I tried it again with a new project but I think the cli was able to catch the change on the package directory this time. If this seems like an intermittent issue we might want to address, I could add some telemetry for now. We could look at the frequency of alternate default `path`s and whether the file is actually created under the specified path after the user retrieves; this should let us see if there's any discrepancy between where the file was intended to be created and where it actually got created.